parser-typechecker/tests/Unison/Test/Codebase/Causal.hs

```diff
@@ -146,19 +146,33 @@ oneCausal = Causal.one (Set.fromList [1])
 -- generateRandomCausals :: Causal Identity Hash (Set Int64)
 -- generateRandomCausals = undefined
 
+easyCombine
+  :: (Monad m, Semigroup d)
+  => (e -> e -> m e)
+  -> (e -> e -> m d)
+  -> (e -> d -> m e)
+  -> (Maybe e -> e -> e -> m e)
+easyCombine comb _    _    Nothing l r = comb l r
```

Comment:
This shows that the new signature of `threeWayMerge` is strictly more powerful than the old - given a `combine`, `diff` (with associated semigroup), and `apply` function, you can construct the argument to `Causal.threeWayMerge`. We considered putting this in `Causal`, but it wouldn't be used anywhere in the codebase, so left it here for now.